### PR TITLE
All the things

### PR DESCRIPTION
**Pull request by Tricopetaurus**
how to even review this

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -1,145 +1,177 @@
...
+    runIntakeBackwardsTrigger.whileTrue(new IntakeCommand(m_intake, false));
+
+    Trigger invertDrivetrainTrigger = m_driverHID.button(1)
+    .onTrue(m_tankDriveSubsystem.new SwitchDrivetrainInvert(m_tankDriveSubsystem));
+
+    // new Intake
+    // Intake is reversed when right bumper is pressed
+    SwitchIntakeDirection m_switchIntake = m_intake.new SwitchIntakeDirection(m_intake);
+    Trigger switchIntakeTrigger = m_driverHID.button(6);
+    switchIntakeTrigger.onTrue(m_switchIntake);
```
Comment (Tricopetaurus) [inline src/main/java/frc/robot/RobotContainer.java]: I believe this section, and really the SwitchIntakeDirection command, are both unneeded now. @AkshayKuchibhatla can you confirm and if so chop this out?

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -1,145 +1,177 @@
...
-    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=FUNCTIONS
-  public CommandGenericHID getController() {
-      return m_driverHID;
+  private void invertDrivetrain() {
+    if (m_tankDriveSubsystem.getInverted()) {
+      m_tankDriveSubsystem.setInverted(false, false, false);
+      System.out.println("Invert Off");
+    } else {
+      m_tankDriveSubsystem.setInverted(true, true, true);
+      System.out.println("Invert On");
```
Comment (Tricopetaurus) [inline src/main/java/frc/robot/RobotContainer.java]: I believe this is redundant with the invert drivetrain command existing within the TankDriveSubsystem.

@tobyMickelson  or @CalebNeal07 (I don't remember who worked the original....) can you confirm?

**Comment by Tricopetaurus** [comment]
Consider replacing the somewhat obscure `.button()` calls in RobotContainer with the more friendly `.a()` or `.b()` etc. We're using the XBoxController and PS4Controller classes now, so we have those conveniences. Makes the code much more readable too.

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -1,145 +1,177 @@
...
-  
+  private final TankDriveSubsystem.driveMotorCommand m_driveCommand = m_tankDriveSubsystem.new driveMotorCommand(
+      rightDriveTrain,
+      leftDriveTrain,
+      m_tankDriveSubsystem);
+
+  private final TankDriveSubsystem.driveMotorCommand m_operatorDrive = m_tankDriveSubsystem.new driveMotorCommand(
+      () -> m_operatorHID.getRawAxis(1),
+      () -> m_operatorHID.getRawAxis(5),
+      m_tankDriveSubsystem);
```
Comment (Tricopetaurus) [inline src/main/java/frc/robot/RobotContainer.java]: I believe @Benito-Esposti / @tobyMickelson decided that giving operator the ability to drive would be more dangerous or confusing than useful. If true, we should remove this code as well as the switch enabling it down around like 99.

**Comment by Tricopetaurus** [comment]
Smattering of TODO notes I have. Up to the team on what priority to rank these, they're just things I don't see in here currently. 

- Intake subsystem needs two additional motors added to it. These are for the star things. The two motors need to run invert of each other. The `.follow()` function can take an optional 2nd arg that's a boolean for "invert" so that's useful. For now I would say just drive that with midtake, doesn't need separate control

- Consider putting up a generic shoot command for testing. In my demo branch I'm just using something like ...
```
        shoot_med.whileTrue(
            new StartEndCommand(
                () -> m_shooter.SetShooter(0.70),
                () -> m_shooter.SetShooter(0),
                m_shooter)
        );
```

- The "flipIntake" thing is going to change again, most likely. We can chat about this more tomorrow. For now, I would recommend just mapping the manual raise / lower features to driver controller. That way all features can be exercised on one controller. 
  - To lower the stress on the intake arm, they've added a spring
  - The spring is great, it should help the motor not be so stressed
  - The downside is that when you've nearly retracted the arm the spring is pulling in the same direction you're going. So your motor goes from fighting a counter force to being pulled along
  - Probably not a huge deal ? If we want it to move smoothly I'd recommend using the CAN Spark Max's built in PID system with position mode. Yay encoders.
    - https://github.com/REVrobotics/SPARK-MAX-Examples/blob/master/Java/Position%20Closed%20Loop%20Control/src/main/java/frc/robot/Robot.java

**Comment by redguy999** [comment]
> Consider replacing the somewhat obscure `.button()` calls in RobotContainer with the more friendly `.a()` or `.b()` etc. We're using the XBoxController and PS4Controller classes now, so we have those conveniences. Makes the code much more readable too.

I'll do this.

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (AkshayKuchibhatla) [reply src/main/java/frc/robot/RobotContainer.java]: I took it out in the recent commit.

File: src/main/java/frc/robot/RobotContainer.java (hunk 2 above)
Comment (AkshayKuchibhatla) [reply src/main/java/frc/robot/RobotContainer.java]: I took it out in the recent commit, along with all variables, functions etc. related to inverting the drivetrain.

File: src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (AkshayKuchibhatla) [reply src/main/java/frc/robot/RobotContainer.java]: Operator drive? _**WHAT OPERATOR DRIVE??**_

**Review by JakeStew81 — APPROVED** [review]

**Review by alexandersondev — COMMENTED** [review]
Worked at comp

**Pull request merged**